common/coin_support/eth.c

```diff
@@ -303,6 +303,9 @@ bool eth_validate_unsigned_txn(eth_unsigned_txn *eth_utxn_ptr, txn_metadata *met
         return false;
     if (memcmp(eth_utxn_ptr->chain_id, &metadata_ptr->network_chain_id, eth_utxn_ptr->chain_id_size[0]) != 0)
         return false;
+    // do not allow transaction with unknown payload
+    if (eth_utxn_ptr->payload_size > 0 && U32_READ_BE_ARRAY(eth_utxn_ptr->payload) != TRANSFER_FUNC_SIGNATURE)
+        return false;
 
     // Check if payload exists, it's a token transfer operation on EVM
     if (eth_utxn_ptr->payload_size > 0 && U32_READ_BE_ARRAY(eth_utxn_ptr->payload) == TRANSFER_FUNC_SIGNATURE) {
```

Comment:
To avoid a situation where a missing check leads to true return, we should replace return true with false at the end of the  function and update the to return true only when all conditions are fulfilled

Reply:
Created an ticket for this refactor: https://app.clickup.com/t/37308523/CHI-2159
We can release the fix as of now.